Source/WebKit/UIProcess/Extensions/gtk/WebExtensionContextGtk.cpp

```
@@ -0,0 +1,703 @@
+        return;
+
+    m_backgroundContentLoadError = createError(Error::BackgroundContentFailedToLoad, nullString(), error);
+    recordErrorIfNeeded(backgroundContentLoadError());
+
+    unloadBackgroundWebView();
+}
+
+void WebExtensionContext::webViewWebContentProcessDidTerminate(WebKitWebView *webView)
+{
+    if (webView == m_backgroundWebView) {
+        unloadBackgroundWebView();
+
+        if (protectedExtension()->backgroundContentIsPersistent())
+            loadBackgroundWebView();
+
+        return;
+    }
+
+#if ENABLE(INSPECTOR_EXTENSIONS)
+    if (isInspectorBackgroundPage(webView)) {
+        webkit_web_view_load_request(m_backgroundWebView.get(), webkit_uri_request_new(inspectorBackgroundPageURL().utf8().data()));
+        return;
+    }
+#endif
+
+    ASSERT_NOT_REACHED();
+}
+
+#if ENABLE(INSPECTOR_EXTENSIONS)
+
+HashSet<Ref<WebProcessProxy>> WebExtensionContext::processes(const API::InspectorExtension& inspectorExtension) const
+{
+    ASSERT(isLoaded());
+    ASSERT(protectedExtension()->hasInspectorBackgroundPage());
+
+    HashSet<Ref<WebProcessProxy>> result;
+
+    RefPtr inspectorProxy = inspector(inspectorExtension);
+    if (!inspectorProxy)
+        return result;
+
+    ASSERT(m_inspectorContextMap.contains(*inspectorProxy));
+
+    const auto& inspectorContext = m_inspectorContextMap.get(*inspectorProxy);
+    if (auto *backgroundWebView = inspectorContext.backgroundWebView.get()) {
+        RefPtr pageProxy = webkitWebViewGetPage(backgroundWebView);
+        result.add(pageProxy->siteIsolatedProcess());
+    }
+
+    return result;
+}
+
+#endif // ENABLE(INSPECTOR_EXTENSIONS)
+
+String WebExtensionContext::declarativeNetRequestContentRuleListFilePath()
+{
+    if (!m_declarativeNetRequestContentRuleListFilePath.isEmpty())
+        return m_declarativeNetRequestContentRuleListFilePath;
+
+    auto directoryPath = storageIsPersistent() ? storageDirectory() : String(FileSystem::createTemporaryDirectory("DeclarativeNetRequest"_s));
```

The `createTemporaryDirectory` API you added can silently fail and return an empty string. This seems not desirable.

---

There's another createTemporaryDirectory API that doesn't accept a directory prefix. It can also fail silently and return an empty string. This seems to be standard behaviour for the filesystem functions